pkg/clients/gcp/gcp.go

```diff
@@ -190,3 +190,47 @@ func getMissingPermissions(expected, actual []string) (missing []string) {
 	}
 	return
 }
+
+// StringValue converts the supplied string pointer to a string, returning the
+// empty string if the pointer is nil.
+func StringValue(v *string) string {
+	if v == nil {
+		return ""
+	}
+	return *v
+}
+
+// Int64Value converts the supplied int64 pointer to an int, returning zero if
+// the pointer is nil.
+func Int64Value(v *int64) int64 {
+	if v == nil {
+		return 0
+	}
+	return *v
+}
+
+// InitializeString initializes s, presumed to be an optional field of a
+// Kubernetes API object's spec per Kubernetes "late initialization" semantics.
+// s is returned unchanged if it is non-nil or from is the empty string,
+// otherwise a pointer to from is returned.
+// https://github.com/kubernetes/community/blob/db7f270f/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required
+// https://github.com/kubernetes/community/blob/db7f270f/contributors/devel/sig-architecture/api-conventions.md#late-initialization
+func InitializeString(s *string, from string) *string {
```

Comment:
I think instead of `from`, `fallback` or another word that means _it's used only in case `is` wasn't usable_ could be a better name.

Reply:
I experimented with renaming `from` to `fallback`, but that's not the full story; we only "fall back" to the supplied string if the string pointer is nil _and_ the supplied string is non-zero. I also considered `InitializeStringIfUnset(s *string, from string)` and disliked that option for the same reason.

I think the best option is to rely on the Godoc and update the signature to `LateInitializeFoo` to make it extra clear that we're applying the Kubernetes "late initialisation" semantics to the type, rather than generically "initialising" it, which could imply we'd just set it no matter what its value was.